kornia/color/colormap.py

```diff
@@ -0,0 +1,188 @@
+    [1.0, 0.2857142857142857, 0.0],
+    [1.0, 0.3015873015873016, 0.0],
+    [1.0, 0.3174603174603174, 0.0],
+    [1.0, 0.3333333333333333, 0.0],
+    [1.0, 0.3492063492063492, 0.0],
+    [1.0, 0.3650793650793651, 0.0],
+    [1.0, 0.3809523809523809, 0.0],
+    [1.0, 0.3968253968253968, 0.0],
+    [1.0, 0.4126984126984127, 0.0],
+    [1.0, 0.4285714285714285, 0.0],
+    [1.0, 0.4444444444444444, 0.0],
+    [1.0, 0.4603174603174603, 0.0],
+    [1.0, 0.4761904761904762, 0.0],
+    [1.0, 0.492063492063492, 0.0],
+    [1.0, 0.5079365079365079, 0.0],
+    [1.0, 0.5238095238095238, 0.0],
+    [1.0, 0.5396825396825397, 0.0],
+    [1.0, 0.5555555555555556, 0.0],
+    [1.0, 0.5714285714285714, 0.0],
+    [1.0, 0.5873015873015873, 0.0],
+    [1.0, 0.6031746031746031, 0.0],
+    [1.0, 0.6190476190476191, 0.0],
+    [1.0, 0.6349206349206349, 0.0],
+    [1.0, 0.6507936507936508, 0.0],
+    [1.0, 0.6666666666666666, 0.0],
+    [1.0, 0.6825396825396826, 0.0],
+    [1.0, 0.6984126984126984, 0.0],
+    [1.0, 0.7142857142857143, 0.0],
+    [1.0, 0.7301587301587301, 0.0],
+    [1.0, 0.746031746031746, 0.0],
+    [1.0, 0.7619047619047619, 0.0],
+    [1.0, 0.7777777777777778, 0.0],
+    [1.0, 0.7936507936507936, 0.0],
+    [1.0, 0.8095238095238095, 0.0],
+    [1.0, 0.8253968253968254, 0.0],
+    [1.0, 0.8412698412698413, 0.0],
+    [1.0, 0.8571428571428571, 0.0],
+    [1.0, 0.873015873015873, 0.0],
+    [1.0, 0.8888888888888888, 0.0],
+    [1.0, 0.9047619047619048, 0.0],
+    [1.0, 0.9206349206349206, 0.0],
+    [1.0, 0.9365079365079365, 0.0],
+    [1.0, 0.9523809523809523, 0.0],
+    [1.0, 0.9682539682539683, 0.0],
+    [1.0, 0.9841269841269841, 0.0],
+    [1.0, 1.0, 0.0],
+]
+
+
+# Generate complete color map
+class AUTUMN(ColorMap):
+    r"""The GNU Octave colormap `autumn`"""
+
+    def __init__(
+        self, num_colors: int = 64, device: torch.device = torch.device('cpu'), dtype: torch.dtype = torch.float32
+    ):
+        super().__init__(base_colormap=_BASE_AUTUMN, num_colors=num_colors, device=device, dtype=dtype)
+
+
+def apply_colormap(input_tensor: Tensor, colormap: ColorMap) -> Tensor:
+    r"""Apply to a gray tensor a colormap.
+
+    Args:
+        input_tensor: the input tensor of a gray image
+        colormap: the colormap desired to be applied to the input tensor
+
+    Returns:
+        Tensor: A RGB tensor with the applied color map into the input_tensor
+
+    Example:
+        >>> input_tensor = torch.tensor([[[0, 1, 2], [25, 50, 63]]])
+        >>> colormap = AUTUMN()
+        >>> apply_colormap(input_tensor, colormap)
+        tensor([[[1.0000, 1.0000, 1.0000],
+                 [1.0000, 1.0000, 1.0000]],
+        <BLANKLINE>
+                [[0.0000, 0.0159, 0.0317],
+                 [0.3968, 0.7937, 1.0000]],
+        <BLANKLINE>
+                [[0.0000, 0.0000, 0.0000],
+                 [0.0000, 0.0000, 0.0000]]])
+    """
+    # TODO: implement to work with a batch of images
+    KORNIA_CHECK_IS_GRAY(input_tensor)
```

Comment:
should we enforce any shape via `KORNIA_CHECK_SHAPE` ?

Reply:
I think `KORNIA_CHECK_IS_GRAY` already do this

Reply:
technically i believe is correct but in terms on semantics tagging e.g a depth map as a grayscale image is wrong.
Maybe we can iterate the check api and more explicit names like `KORNIA_CHECK_SINGLE_CHANNEL`  or similar which accepts `(("B", None), "H", "W")`

Reply:
A doubt, your understanding for these cases be the same thing? `(("B", None), "H", "W")` or `("*", "H", "W")`
or the `*` here means we can have tensors with shape like `(2, 2, 2, H, W)`?